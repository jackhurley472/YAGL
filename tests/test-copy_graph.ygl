Graph g;
Node n1("Hi there");
Node n2("hey");
Node n3("Bye");
g: + n1 + n2 + n3 -> n2 <-5 n1;
Graph g2;
g2 = copy_graph_lib(g);
print("original:");
print(g);
print("Copied:");
print(g2);
Graph copy_graph_lib(Graph g) {

	Graph g2;

	int nodes = g.num_nodes;
	int curr = 0;
	while (curr < nodes) {
		Node current;
		current = g.node[curr];
		g2: + current;
		curr = curr + 1;
	}
	
	curr = 0;
	while (curr < nodes) {

		Node current = g.node[curr];
		int neighs = g.num_neighbors[current];
		int on = 0;
		while (on < neighs) {
			Node to;
			to  = g.neighbor[current, on];
			int val = g.weight[current, to];
			g2: current ->val to;
			on = on + 1;
		} 

		curr = curr + 1;
	}

	return g2;

}
