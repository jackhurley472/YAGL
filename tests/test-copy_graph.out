original:
============== Graph Print ===============
	Stats: 	4	3	2

Nodes:
Node (0) : Hi there --- Node (1) : hey --- Node (2) : Bye


Edges:
[(2) : Bye] ---(1)---> [(1) : hey]
[(0) : Hi there] ---(5)---> [(1) : hey]
============ End Graph Print =============
Copied:
============== Graph Print ===============
	Stats: 	4	3	2

Nodes:
Node (0) : Hi there --- Node (1) : hey --- Node (2) : Bye


Edges:
[(0) : Hi there] ---(5)---> [(1) : hey]
[(2) : Bye] ---(1)---> [(1) : hey]
============ End Graph Print =============
